.github/workflows/ci-validation.yml

```diff
@@ -35,9 +35,14 @@ jobs:
     - name: Build (Ubuntu 20.04)
       run: |
         sudo cp -rf . /p4c/
-        (cd /p4c/ && sudo -E tools/ci-build.sh)
-        sudo cp -rf /p4c/.ccache .
+        (cd /p4c/ && sudo -E tools/ci-build.sh) || echo "BUILD_SUCCESS=false" >> $GITHUB_ENV
 
     - name: Validate
-      run: sudo -E ctest  -R toz3-validate-p4c --output-on-failure --schedule-random
-      working-directory: /p4c/build
+      if: env.BUILD_SUCCESS == 'true'
+      run: |
+          docker run --privileged -w /p4c/build -e $CTEST_PARALLEL_LEVEL p4c ctest  -R toz3-validate-p4c --output-on-failure --schedule-random
```

Comment:
Are you sure? Why docker again?

Reply:
Something broke with this file while rebasing. Will fix.